File: source/backend/python/application/port/ingress/send_money_command.py
from application.domain.model.account import Account
from application.domain.model.money import Money
from common import validation

class SendMoneyCommand:
    def __init__(self, source_account: Account, target_account: Account, money: Money):
        self.source_account = source_account
        self.target_account = target_account
        self.money = money
        validation.SendMoneyCommand(source_account, target_account, money)

File: source/backend/typescript/application/port/in/SendMoneyCommand.ts
/**
 * SendMoneyServiceの入力モデル
 * 入力値の妥当性確認が責務
 */

import { Account } from "../../domain/model/Account";
import { Money } from "../../domain/model/Money"
import { Validation } from "../../../common/Validation";

export class SendMoneyCommand{
    public readonly sourceAccount: Account;
    public readonly targetAccount: Account;
    public readonly money: Money;

    constructor(sourceAccount: Account, targetAccount: Account, money: Money){
        this.sourceAccount = sourceAccount;
        this.targetAccount = targetAccount;
        this.money = money;
        Validation.sendMoneyCommand(this);
    }
}